File: app/src/main/java/me/zheteng/android/videosaver/parser/extractor/MiaopaiExtractor.java
package me.zheteng.android.videosaver.parser.extractor;

import android.support.annotation.NonNull;
import me.zheteng.android.videosaver.parser.Extractor;
import me.zheteng.android.videosaver.parser.Video;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * 秒拍抽取
 */
public class MiaopaiExtractor implements Extractor {
    @NonNull
    @Override
    public List<Video> extract(String html) {
        Pattern pattern = Pattern.compile("\"videoSrc\":\"([^\"]+)\"");
        Matcher matcher = pattern.matcher(html);
        List<Video> list = new ArrayList<>();
        if (matcher.find()) {
            list.add(new Video(matcher.group(1)));
        }
        return list;
    }
}

File: app/src/main/kotlin/me/zheteng/android/videosaver/parse/extractor/MiaopaiExtractor.kt
package me.zheteng.android.videosaver.parse.extractor

import me.zheteng.android.videosaver.parse.Extractor
import me.zheteng.android.videosaver.parse.Video
import java.util.*
import java.util.regex.Pattern


/**
 * 秒拍抽取
 */
class MiaopaiExtractor : Extractor {
    override fun extract(html: String): List<Video> {
        val pattern = Pattern.compile("\"videoSrc\":\"([^\"]+)\"")
        val matcher = pattern.matcher(html)
        val list = ArrayList<Video>()
        if (matcher.find()) {
            list.add(Video(matcher.group(1)))
        }
        return list
    }
}
